src/main/java/io/fusionauth/samlv2/service/DefaultSAMLv2Service.java

```diff
@@ -1042,6 +1061,180 @@ private AlgorithmParameterSpec createAlgorithmParameterSpec(EncryptionAlgorithm
     }
   }
 
+  /**
+   * Decrypt an encrypted XML element according to the XML Encryption spec and optionally verify the signature.
+   *
+   * @param encryptedAssertion     an {@code EncryptedElement} containing an encrypted assertion
+   * @param transportEncryptionKey a private key used to decrypt the symmetric key used to decrypt the assertion
+   * @param verifySignature        if {@code true}, the method will verify an embedded signature. Failure to locate or
+   *                               verify the signature will result in an exception.
+   * @param signatureKeySelector   a key
+   * @return the decrypted {@code Assertion} element
+   * @throws SAMLException if there was an issue decrypting the {@code EncryptedElement} to a SAML {@code Assertion}
+   *                       element
+   */
+  private AssertionType decryptAssertion(EncryptedElementType encryptedAssertion, PrivateKey transportEncryptionKey,
+                                         boolean verifySignature, KeySelector signatureKeySelector)
+      throws SAMLException {
+    // Extract the encrypted assertion encryption key from the XML
+    EncryptedKeyType encryptedKey = extractEncryptedAssertionEncryptionKey(encryptedAssertion);
+
+    // Determine the assertion EncryptionAlgorithm.
+    var assertionEncryptionAlgorithmUri = encryptedAssertion.getEncryptedData().getEncryptionMethod().getAlgorithm();
+    var assertionEncryptionAlgorithm = EncryptionAlgorithm.fromURI(assertionEncryptionAlgorithmUri);
+    if (assertionEncryptionAlgorithm == null) {
+      throw new SAMLException("Unable to determine assertion encryption algorithm from URI [" + assertionEncryptionAlgorithmUri + "]");
+    }
+
+    // Decrypt the assertion encryption key using the transport key
+    Key assertionEncryptionKey;
+    try {
+      assertionEncryptionKey = decryptKey(encryptedKey, transportEncryptionKey, assertionEncryptionAlgorithm);
+    } catch (NoSuchPaddingException | NoSuchAlgorithmException | InvalidAlgorithmParameterException |
+             InvalidKeyException | IllegalBlockSizeException | BadPaddingException | InvalidKeySpecException e) {
+      throw new SAMLException("Unable to decrypt symmetric key using transport key", e);
+    }
+
+    // Decrypt the assertion using the decrypted symmetric key
+    byte[] assertionBytes;
+    try {
+      assertionBytes = decryptElement(encryptedAssertion.getEncryptedData().getCipherData().getCipherValue(), assertionEncryptionAlgorithm, assertionEncryptionKey);
+    } catch (NoSuchPaddingException | NoSuchAlgorithmException | InvalidAlgorithmParameterException |
```

Comment:
Do we just want to catch a `GeneralSecurityException` here? Looks like that would cover everything listed, unless you think that is a bad idea.